packages/data-sdk/src/Device.ts

```diff
@@ -245,81 +243,122 @@ export class Device extends EventEmitter implements IRealtimeDevice {
 
   /**
    * Starts a real-time connection with the remote device using WebRTC.
-   * @param {number} [sessionType] - Optional session type to be used for the connection.
+   * @param {number} [options] - Optional session type to be used for the connection.
    * @throws `Error`  If the connection could not be established or if a connection already exists.
    * @returns {void}
    */
+  async startRealtimeConnection(
+    options: SessionType | IStartRealtimeConnectionOptions = {}
+  ): Promise<void> {
+    console.debug(`${new Date().toISOString()} :: Connection start requested`);
 
-  async startRealtimeConnection(sessionType?: SessionType): Promise<void> {
-    if (!this.rtcClient || this.connectionMonitorInterval === undefined) {
-      if (this.rtcClient) {
-        console.error(
-          "overwriting existing rtcClient due to missing connectionMonitorInterval"
-        );
-      }
+    if (this.rtcClient && this.connectionMonitorInterval !== undefined) {
+      throw new Error(
+        `Already created realtime connection to device ${this.id}`
+      );
+    }
 
-      let rtcClient;
-
-      if (rtcClientVersion === "1") {
-        rtcClient = new RtcClientV1({
-          signalingClient: new RtcSignalingClient(
-            FORMANT_API_URL + "/v1/signaling"
-          ),
-          getToken: async () =>
-            defined(
-              Authentication.token,
-              "Realtime when user isn't authorized"
-            ),
-          receive: this.handleMessage,
-        });
-      } else {
-        const pool = sessionType
-          ? AppRtcClientPools[sessionType]
-          : defaultRtcClientPool;
-        rtcClient = pool.get(this.handleMessage);
-      }
+    if (this.rtcClient) {
+      console.warn(
+        "overwriting existing rtcClient due to missing connectionMonitorInterval"
+      );
+    }
 
+    const {
+      sessionType,
+      deadlineMs = 10_000,
+      maxConnectRetries = 3,
+    } = typeof options === "number"
+      ? ({ sessionType: options } as IStartRealtimeConnectionOptions)
+      : options;
+
+    const pool = getRtcClientPool({
+      sessionType,
+      version: getRtcClientVersion() ?? "2",
+    });
+    const rtcClient = pool.get(this.handleMessage);
+
+    let cancelled = false;
+    const deadlinePromise = new Promise<never>((_, reject) =>
```

Comment:
_Probably_ doesn't matter in the grand scheme of things. But I had implemented this deadline functionality because of some weirdness with gRPC (web-grpc specifically). If the proxy is down (as is typical during deploys), then gRPC requests could end up never resolving.

I don't recall what our V1 timeout is set to, but the deadline may not be needed as a result. That being said, I think 10s is more than reasonable as a hard timeout.

Reply:
yea, to me, _for this_—_i.e._ a series of "blocking" loops—a temporal timeout makes more sense to me than specify retry loops. So, as a consumer, I dont really care what you gotta do, but if you can't verify this connection by 𝑛 milliseconds, then stop what you are doing and I'll sort it out. But I can totally see cases where certain customers, might prefer longer (or shorter) thresholds for that. Like I said, that makes the "most" sense to me. 🤞 So, for the underlying api, those timeouts, _per call_, I think of _"different"_, i guess.